alf/environments/utils.py

```diff
@@ -19,33 +19,40 @@
 
 from alf.environments import suite_mario
 from alf.environments import suite_socialbot
+from alf.environments import suite_dmlab
 
 
 @gin.configurable
 def create_environment(env_name='CartPole-v0',
                        env_load_fn=suite_gym.load,
-                       num_parallel_environments=30):
+                       num_parallel_environments=30,
+                       force_unwrapped=False):
     """Create environment.
 
     Args:
         env_name (str): env name
         env_load_fn (Callable) : callable that create an environment
         num_parallel_environments (int): num of parallel environments
+        force_unwrapped (bool): force to create a single env in the current
+            process; use only when you are certain that multiple envs in the
+            current process can co-exist.
     Returns:
         TFPyEnvironment
     """
-    xarg = {}
-    if env_load_fn in (suite_socialbot.load, env_load_fn == suite_mario.load):
-        # No need to wrap with process if num_parallel_environments > 1,
-        # because ParallelPyEnvironment will do that.
-        xarg = dict(wrap_with_process=num_parallel_environments == 1)
+    wrappable = (env_load_fn in (suite_socialbot.load, suite_mario.load,
+                                 suite_dmlab.load))
 
-    if num_parallel_environments == 1:
+    xarg = dict()
+    if force_unwrapped or num_parallel_environments == 1:
+        if wrappable:
+            xarg = dict(wrap_with_process=not force_unwrapped)
+        # for unwrappable envs, there is no arg called "wrap_with_process"
         py_env = env_load_fn(env_name, **xarg)
     else:
         py_env = parallel_py_environment.ParallelPyEnvironment(
```

Comment:
should not wrap for num_parallel_environments > 1

Reply:
fixed

Reply:
I have reverted the change of moving the global flag inside load(). The reason is that load might be called by ParallelPyEnvironment with wrap_with_process=False. In this case, there is no env created in the main process but the global flag will be set to True. The flag should really be set only when load() is called in the first branch in create_environment()